front/lib/api/assistant/conversation.ts

```diff
@@ -175,6 +183,120 @@ export async function* postUserMessage(
+  const agentMessages: AssistantAgentMessageType[] = [];
+
+  await front_sequelize.transaction(async (t) => {
+    let nextMessageRank =
+      ((await AssistantMessage.max<number | null, AssistantMessage>("rank", {
+        where: {
+          assistantConversationId: conversation.id,
+        },
+        transaction: t,
+      })) ?? -1) + 1;
+
+    const userMessageRow = await AssistantMessage.create(
+      {
+        sId: generateModelSId(),
+        rank: nextMessageRank++,
+        assistantConversationId: conversation.id,
+        parentId: null,
+        assistantUserMessageId: (
+          await AssistantUserMessage.create(
+            {
+              message: message,
+              userContextUsername: context.username,
+              userContextTimezone: context.timezone,
+              userContextFullName: context.fullName,
+              userContextEmail: context.email,
+              userContextProfilePictureUrl: context.profilePictureUrl,
+              userId: user ? user.id : null,
+            },
+            { transaction: t }
+          )
+        ).id,
+      },
+      {
+        transaction: t,
+      }
+    );
+
+    userMessage = {
+      id: userMessageRow.id,
+      sId: userMessageRow.sId,
+      type: "user_message",
+      visibility: "visible",
+      version: 0,
+      parentMessageId: null,
+      user: user,
+      mentions: mentions,
+      message: message,
+      context: context,
+    };
+
+    // for each assistant mention, create an "empty" agent message
+    for (const m of mentions) {
+      if (isAssistantAgentMention(m)) {
+        const agentMessageRow = await AssistantMessage.create(
+          {
+            sId: generateModelSId(),
+            rank: nextMessageRank++,
+            assistantConversationId: conversation.id,
+            parentId: userMessage.id,
+            assistantAgentMessageId: (
+              await AssistantAgentMessage.create({}, { transaction: t })
+            ).id,
+          },
+          {
+            transaction: t,
+          }
+        );
+        agentMessages.push({
+          id: agentMessageRow.id,
+          sId: agentMessageRow.sId,
+          type: "agent_message",
+          visibility: "visible",
+          version: 0,
+          parentMessageId: userMessage.sId,
+          status: "created",
+          action: null,
+          message: null,
+          feedbacks: [],
+          error: null,
+          configuration: {
+            sId: m.configurationId,
+            status: "active",
+            name: "foo", // TODO
+            pictureUrl: null, // TODO
+            action: null, // TODO
+            message: null, // TODO
+          },
+        });
+      }
+    }
+  });
+
+  if (!userMessage) {
+    throw new Error("Unreachable.");
+  }
+  yield {
+    type: "user_message_new",
+    message: userMessage,
+  };
+
+  for (const e of agentMessages) {
+    yield {
+      type: "agent_message_new",
+      message: e,
+      created: Date.now(),
```

Comment:
@spolu  do we really need this here ? Or should it rather be a property of the messages (both user messages and assistant messages)